src/dmd/semantic3.d

```diff
@@ -1000,6 +1000,28 @@ private extern(C++) final class Semantic3Visitor : Visitor
                 if (global.params.useOut == CHECKENABLE.off)
                     fens = null;
             }
+
+            // if the function body or the in contracts contain "assert(__ctfe)" set the compile time only flag
+            {
+                if (funcdecl.fbody)
+                {
+                    if (auto cs = funcdecl.fbody.isCompoundStatement())
+                        if (auto s = cs.statements)
+                            if (s.length && isAssertCtfe((*s)[0]))
```

Comment:
Why the parens around `*s`?

Reply:
because s is a pointer to an array. not having the `*` there would be i'd be slicing the pointer rather than indexing the array.